improver_tests/spotdata/test_HeightAdjustment.py

```diff
@@ -0,0 +1,267 @@
+    )
+    return neighbour_cube
+
+
+@pytest.mark.parametrize(
+    "cube_name, expected",
+    (
+        (
+            "prob_cube",
+            [
+                [0.1, 0.7, 1],
+                [0, 0, 0.35555556],
+                [0.2, 0.3, 0.4],
+                [0.23888889, 0.27777778, 0.9],
+            ],
+        ),
+        (
+            "prob_cube_realizations",
+            [
+                [
+                    [0.1, 0.7, 1],
+                    [0, 0, 0.35555556],
+                    [0.2, 0.3, 0.4],
+                    [0.23888889, 0.27777778, 0.9],
+                ],
+                [
+                    [0.1, 0.7, 1],
+                    [0, 0, 0.35555556],
+                    [0.2, 0.3, 0.4],
+                    [0.23888889, 0.27777778, 0.9],
+                ],
+            ],
+        ),
+        (
+            "realization_cube",
+            [
+                [1000, 3671.91601, -200.0, 428.08399],
+                [3000.0, 9671.91601, 0.0, 528.08399],
+                [4000.0, 10671.91601, 30.0, 478.08399],
+            ],
+        ),
+        (
+            "percentile_cube",
+            [[1400, 3900, -500, 200], [2000, 9900, 0, 200], [3000, 10900, 30, 250]],
+        ),
+    ),
+)
+def test_cube(cube_name, expected, neighbour_cube, request):
+    """Tests plugin produces correct metadata and results for different input cubes"""
+    cube = request.getfixturevalue(cube_name)
+    result = SpotHeightAdjustment()(cube, neighbour_cube)
+    coords_original = [c.name() for c in cube.dim_coords]
+    coords_result = [c.name() for c in result.dim_coords]
+
+    assert coords_original == coords_result
+    assert result.units == cube.units
+    assert result.name() == cube.name()
+    if cube_name == ("prob_cube" or "prob_cube_realizations"):
+        assert result.coord(
+            "cloud_base_height_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas"
+        ) == cube.coord(
+            "cloud_base_height_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas"
+        )
+    np.testing.assert_allclose(result.data, expected)
+
+
+def test_prob_cube_threshold_unit(prob_cube, neighbour_cube):
+    """Tests that correct units and data are returned if probability cubes threshold is converted
+    to metres"""
+    prob_cube.coord(
+        "cloud_base_height_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas"
+    ).units = "km"
+    expected = [
+        [0.1, 0.7, 1.0],
+        [0, 0, 0.3999556],
+        [0.2, 0.3, 0.4],
+        [0.1002, 0.2000778, 0.9],
+    ]
+
+    result = SpotHeightAdjustment()(prob_cube, neighbour_cube)
+    assert (
+        prob_cube.coord(
+            "cloud_base_height_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas"
+        ).units
+        == result.coord(
+            "cloud_base_height_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas"
+        ).units
+    )
```

Comment:
The units of the threshold coordinate on the outputted cube should be the same as the units of the threshold coordinate on the inputted cube. In other words while the plugin does need the units to be convertable to metres it shouldn't actually be changin the units on the cube. 

In the plugin, the units of spot_cube's threshold coordinate are never actually changes on the cube but the threshold coordinate is extracted and the value of the points on the coordinate are changed so the cube doesn't need to be converted back to metres. I might just be being paronoid by checking this here and in test_prob_cube_threshold_unit but I wanted to make sure the plugin wasn't altering the units of the cube.

Reply:
After our discussion I've now updated the plugin so that the threshold coordinate is copied before the units are changed. 

This did also highlight a problem with a lot of my tests that because the cube was being altered in the plugin I was essentially comparing the cube metadata to itself as apposed to itself pre-plugin which is why the tests didn't pick this up. I've now updated the tests so the errors get caught.